src/layers/repository/product_repo.py

```diff
@@ -0,0 +1,175 @@
+from domain.core.product import (
+    Product,
+    ProductCreatedEvent,
+    ProductDeletedEvent,
+    ProductKey,
+    ProductKeyAddedEvent,
+    ProductKeyRemovedEvent,
+    ProductUpdatedEvent,
+    Relationship,
+    RelationshipAddedEvent,
+    RelationshipRemovedEvent,
+)
+from repository.keys import ods_pk, product_key_sk, product_pk, product_relationship_sk
+
+from .repository import NotFoundException, Repository, TooManyFoundException
+from .utils import marshall, unmarshall
+
+
+class ProductRepository(Repository[Product]):
+    def __init__(self, table_name):
+        super().__init__(table_name=table_name, model=Product)
+
+    def handle_ProductCreatedEvent(self, event: ProductCreatedEvent, entity: Product):
+        return {
+            "Put": {
+                "TableName": self.table_name,
+                "Item": marshall(
+                    {
+                        "pk": product_pk(event.id),
+                        "sk": product_pk(event.id),
+                        "pk_1": ods_pk(event.ods_code),
+                        "sk_1": product_pk(event.id),
+                        "name": event.name,
+                        "type": event.type,
+                        "ods_code": event.ods_code,
+                        "product_team_id": event.product_team_id,
+                        "status": event.status,
+                    }
```

Comment:
worth creating a dataclass for these as feels a bit icky having the data model written freehand